### DISCUSSION TITLE
What is the purpose of following code?

### DISCUSSION THREAD

**Discussion by pavolkacej** [suragnair/alpha-zero-general#267] [Q&A]
https://github.com/suragnair/alpha-zero-general/blob/master/Coach.py#L69
Shouldn't it be better to simplify the operation to multiple lines? This way it is really hard to read. I understand `x[0], x[2]`, but what about that `r * ...`? What does the computed value do, and how should this computation work?

I know this oneliners may be fast, but I suggest it is better to do something following

```
finalList = []
for x in trainExamples:

    # compute 1st part of magical value
    # compute 2nd part of magical value
    # compute 3nd part of magical value
    finalList.append( [x[0], x[2], magicalValue] )

return finalList
```

Using multiple lines would make it so much easier to follow. If code works, that is great, but please think about other people, that will have to read it in the future.

**Comment by pavolkacej** [comment]
@suragnair

**Comment by adamlawson22** [comment]
Yeah I agree that this could be made easier to read.
I believe the "magicalValue" is something to do with the reward.
